### PR TITLE
Changed sftp_handle to download everything in the output location.

### PR DESCRIPTION
**Pull request by Shettland**
To fit the goal of the title I did some changes in:
1. [configuration.json](https://github.com/BU-ISCIII/relecov-tools/blob/develop/relecov_tools/conf/configuration.json): changed "storage_local_folder" field name in "stfp_handle" and removed "tmp_metadata_folder" as it has no use now. I also found out some bugs related to wrong column names in "metadata_lab_heading" field so i deleted them too.
2. [sftp_handle.py](https://github.com/BU-ISCIII/relecov-tools/blob/develop/relecov_tools/sftp_handle.py): I deleted "metadata_tmp_folder" as everything is now going to be downloaded and processed in the same location. Also adapted self.storage_local_folder to the new field name in config file. In the process I encountered some bugs related to ad-hoc expected positions in get_sample_fastq_file_names method so I implemented many a few variables to make the code a little more flexible. 

I also reestructured [long_table_parse.py](https://github.com/BU-ISCIII/relecov-tools/blob/develop/relecov_tools/long_table_parse.py)'s parse_file method in order to reduce some lines of code.

Finally, I changed the -help flag info in download's --output_location inside [__main__.py](https://github.com/BU-ISCIII/relecov-tools/blob/develop/relecov_tools/__main__.py) to make it a little bit more informative.

### CODE REVIEW

File: relecov_tools/long_table_parse.py [hunk 1]
```diff
@@ -83,47 +83,49 @@ def parse_file(self):
...
 
-        self.validate_file(lines[0].strip().split(","))
         stderr.print("[green]Successful checking heading fields")
         stderr.print("[blue]Parsing the input file")
         heading_index = {}
         headings_from_csv = lines[0].strip().split(",")
         for heading in self.long_table_heading:
             heading_index[heading] = headings_from_csv.index(heading)
 
+        aux_dict = {
```
Comment (saramonzon) [inline relecov_tools/long_table_parse.py:96]: As you told me this could be set in the config file right?

File: relecov_tools/sftp_handle.py [hunk 2]
```diff
@@ -391,9 +413,9 @@ def validate_metadata_file(self, fetched_folder):
             stderr.print("[red] Too many metadata files on " + fetched_folder)
             return False
         target_meta_file = os.path.join(fetched_folder, meta_files[0])
-        temp_folder = self.metadata_tmp_folder
-        os.makedirs(temp_folder, exist_ok=True)
-        local_meta_file = os.path.join(temp_folder, os.path.basename(target_meta_file))
+        out_folder = self.storage_local_folder
+        os.makedirs(out_folder, exist_ok=True)
```
Comment (saramonzon) [inline relecov_tools/sftp_handle.py:417]:
```suggestion
        os.makedirs(self.storage_local_folder, exist_ok=True)
```

File: relecov_tools/sftp_handle.py [hunk 3]
```diff
@@ -391,9 +413,9 @@ def validate_metadata_file(self, fetched_folder):
...
+        local_meta_file = os.path.join(out_folder, os.path.basename(target_meta_file))
```
Comment (saramonzon) [inline relecov_tools/sftp_handle.py:418]:
```suggestion
        local_meta_file = os.path.join(self.storage_local_folder, os.path.basename(target_meta_file))
```

File: relecov_tools/sftp_handle.py [hunk 4]
```diff
@@ -412,16 +434,16 @@ def validate_metadata_file(self, fetched_folder):
     def validate_fetched_files(self, fetched_folder):
         """Check if the files in the fetched folder are the ones defined in metadata file"""
         local_meta_file = self.validate_metadata_file(fetched_folder)
-        temp_folder = self.metadata_tmp_folder
+        out_folder = self.storage_local_folder
```
Comment (saramonzon) [inline relecov_tools/sftp_handle.py:437]:
```suggestion
```

File: relecov_tools/sftp_handle.py [hunk 5]
```diff
@@ -412,16 +434,16 @@ def validate_metadata_file(self, fetched_folder):
...
-            log.error("Excel file for metadata not found %s", fetched_folder)
+            log.error("Excel file for metadata not found in %s", fetched_folder)
             stderr.print(
                 "[red] Metadata file could not be obtained from " + fetched_folder
             )
             return False
         allowed_extensions = self.allowed_sample_ext
         samples_files_list = self.get_sample_fastq_file_names(
-            temp_folder, local_meta_file
+            out_folder, local_meta_file
```
Comment (saramonzon) [inline relecov_tools/sftp_handle.py:446]:
```suggestion
            self.storage_local_folder, local_meta_file
```

File: relecov_tools/sftp_handle.py [hunk 6]
```diff
@@ -451,7 +474,7 @@ def validate_fetched_files(self, fetched_folder):
                     str(mismatch_files[0:9]),
                 )
             log.error(f"List of files that mismatch: {str(mismatch_files)}")
-            self.delete_local_folder(temp_folder)
+            self.delete_local_folder(out_folder)
```
Comment (saramonzon) [inline relecov_tools/sftp_handle.py:477]:
```suggestion
            self.delete_local_folder(self.storage_local_folder)
```

**Review by saramonzon — APPROVED** [review]
I would also change the `output_location` to `platform_storage_folder` so variables for the same thing are named the same :) 

Code looks pretty nice, thanks and congratulations!

File: relecov_tools/long_table_parse.py (hunk 1 above)
Comment (Shettland) [reply relecov_tools/long_table_parse.py:96]: Correct, I kept it there just as a temporary option but I'll add it to the config file then.

File: relecov_tools/sftp_handle.py [hunk 7]
```diff
@@ -79,12 +76,6 @@ def __init__(
                     self.storage_local_folder = config_json.get_topic_data(
                         "sftp_handle", "storage_local_folder"
```
[the remark this answers is not in the document]
Comment (Shettland) [reply relecov_tools/sftp_handle.py:77]: You are right, I'll change the name of the variables to make it more consistent. And yes, those should be all "platform_storage_folder"

**Comment by Shettland** [comment]
I introduced the changes that were mentioned previously.

**Pull request merged**